### PR TITLE
Add process namespace sharing for ingress controller

### PR DESCRIPTION
**Pull request by panzouh**
### Proposed changes

The changes allow to configure process namespace sharing for a pod.
 
Solves : #4558

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: charts/nginx-ingress/values.yaml [hunk 1]
```diff
@@ -64,6 +64,9 @@ controller:
   ## Enables debugging for NGINX. Uses the nginx-debug binary. Requires error-log-level: debug in the ConfigMap via `controller.config.entries`.
   nginxDebug: false
 
+  ## Shared process namespace between containers in the Ingress Controller pod.
+  sharedProcessNamespace: false
```
Comment (lucacome) [inline charts/nginx-ingress/values.yaml:68]: should we name it `share` instead of `shared`?

Comment (panzouh) [reply charts/nginx-ingress/values.yaml:68]: The attribute in the pod template is indeed `shareProcessNamespace` i thought it was more logic but i don't mind changing it

Comment (panzouh) [reply charts/nginx-ingress/values.yaml]: > should we name it `share` instead of `shared`?

Updated on 8961b32f8f4a1a62883ba28fa92d8c88f657bf61

**Comment by panzouh** [comment]
> @panzouh there are a few things missing from this PR.
> 
> 1. Docs will need to be updated for this new value. You can add a new entry for this in `charts/nginx-ingress/README.md`. This is to ensure we provide a breakdown of what the setting does.
> 2. The `values.json.schema` file in `charts/nginx-ingress/` will need to be updated as well.

Of course, given the fact that these modifications are not described in the contribute guide nor in the precommit config how it should be done ?

**Review by lucacome — APPROVED** [review]
@panzouh LGTM, but seems like you need to rebase your branch.

**Review by shaun-nx — APPROVED** [review]

**Comment by shaun-nx** [comment]
HI @panzouh was having some problems with the pipeline. I'm re-running the failed jobs now. I will check back in on this PR in my morning (bit late as I'm checking this 😅)

**Comment by shaun-nx** [comment]
Thanks for adding this @panzouh ! Keep us informed on https://github.com/nginxinc/kubernetes-ingress/issues/4558 is this setting helps you!

**Pull request merged**